lib/identity/parsers/gc_ids.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require_relative 'gc_ids_constants'
+require_relative 'gc_ids_helper'
+
+module Identity
+  module Parsers
+    module GCIds
+      include Identity::Parsers::GCIdsConstants
+      include Identity::Parsers::GCIdsHelper
+
+      # @param ids [Array] An array of XML objects representing ids to parse
+      # @return [Hash] An hash representing the parsed ids
+      def parse_xml_gcids(ids)
+        return unless ids.is_a? Array
+
+        ids = ids.map(&:attributes)
+
+        birls_ids = select_ids(select_extension(ids, BIRLS_IDS_REGEX, VA_ROOT_OID)) || []
+
+        {
```

Comment:
Is it possible to build a data structure that we can iterate over and reduce the number of lines here?

```
parsed_ids = {}
REGEX_MAPPINGS.each do | k, v|
  ids = select_ids(select_extension(ids, v[:pattern], v[:root])
  if v[:first] == true 
    parsed_ids[k] = ids&.first
  else
    parsed_ids[k] = ids
  end
end
```

Reply:
Cool, check it out now, I took this suggestion. Had to move a couple things around to make this more clear but should still be good